microsoft/testsuites/cvm/cvm_attestation_tool.py

```
@@ -1,18 +1,140 @@
+            expected_exit_code_failure_message=failure_msg,
+            cwd=self.snp_report_tool_path,
+            shell=True,
+            sudo=True,
+            force_run=True,
+        )
+
+        output: str = command.stdout
+        report = jwt.decode(
+            output, algorithms=["RS256"], options={"verify_signature": False}
+        )
+
+        self._save_attestation_report(json.dumps(report), log_path=log_path)
+
+        self._save_kernel_logs(log_path=log_path)
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        tool_path = self.get_tool_path(
+            use_global=True,
+        )
+
+        self.repo_root = tool_path / "confidential-computing-cvm-guest-attestation"
+        self.snp_report_tool_path = self.repo_root / "cvm-attestation-sample-app"
+        self.azguestattestaion = (
+            "https://packages.microsoft.com/repos/"
+            "azurecore/pool/main/a/azguestattestation1/"
+            "azguestattestation1_1.0.5_amd64.deb"
+        )
+        self.cmd_path = self.snp_report_tool_path / "AttestationClient"
+
+    def _install(self) -> bool:
+        self._install_dep_packages()
+        git = self.node.tools[Git]
+        wget = self.node.tools[Wget]
+
+        root_path = self.get_tool_path(
+            use_global=True,
+        )
+
+        git.clone(self.repo, root_path)
+        self.node.execute("apt-get update", sudo=True)
+        deb_package = wget.get(
+            url=self.azguestattestaion,
+        )
+        self.node.execute(f"dpkg -i {deb_package}", sudo=True)
+        self.node.execute(
+            "cmake .",
+            shell=True,
+            cwd=self.snp_report_tool_path,
+        )
+        make = self.node.tools[Make]
+        make.make("", cwd=self.snp_report_tool_path)
+
+        return self._check_exists()
+
+    def _install_dep_packages(self) -> None:
+        # Installing packages supported by Ubuntu
+        posix_os: Posix = cast(Posix, self.node.os)
```

if only support on ubuntu, we need skip on other distro

---

Skip should be in test case level only. In tool, it should raise unsupported distro exception.